jib-core/src/test/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilderTest.java

```diff
@@ -183,6 +183,92 @@ public void testToBlob_reproducibility() throws IOException {
     Assert.assertThat(layerContent, CoreMatchers.is(reproducedLayerContent));
   }
 
+  @Test
+  public void testBuild_parentDirBehavior() throws IOException {
+    Path testRoot = temporaryFolder.getRoot().toPath();
+
+    // the path doesn't really matter on source files, but these are structured
+    Path parent = Files.createDirectories(testRoot.resolve("parent"));
+    Path file = Files.createFile(parent.resolve("file"));
+    Path ignoredParent = Files.createDirectories(testRoot.resolve("ignoredParent"));
+    Path file2 = Files.createFile(ignoredParent.resolve("file2"));
+    Path file3 =
+        Files.createFile(
+            Files.createDirectories(testRoot.resolve("absentParent")).resolve("file3"));
+
+    Blob layer =
+        new ReproducibleLayerBuilder(
+                ImmutableList.of(
+                    new LayerEntry(
+                        parent,
+                        AbsoluteUnixPath.get("/root/parent"),
+                        FilePermissions.fromOctalString("111"),
+                        Instant.ofEpochMilli(1111)),
+                    new LayerEntry(
+                        file,
+                        AbsoluteUnixPath.get("/root/parent/file"),
+                        FilePermissions.fromOctalString("222"),
+                        Instant.ofEpochMilli(2222)),
+                    new LayerEntry(
+                        file2,
+                        AbsoluteUnixPath.get("/root/ignoredParent/file2"),
+                        FilePermissions.fromOctalString("333"),
+                        Instant.ofEpochMilli(3333)),
+                    new LayerEntry(
+                        parent,
+                        AbsoluteUnixPath.get("/root/ignoredParent"),
+                        FilePermissions.fromOctalString("444"),
+                        Instant.ofEpochMilli(4444)),
+                    new LayerEntry(
+                        file3,
+                        AbsoluteUnixPath.get("/root/absenteeParent/file3"),
+                        FilePermissions.fromOctalString("555"),
+                        Instant.ofEpochMilli(5555))))
+            .build();
+
+    Blob equivalentLayer =
```

Comment:
It's possible I took the wrong approach to the test, this is pretty hard to read, perhaps I should verify each entry by iterating over the list of entries after writing to a tarfile (I didn't do this because it does things in a crazy alphabetical order)